File: Cola/Cola.cs
// Cola - Primero que entra, primero que sale

public class Cola<T> 
{
    // Declaro la lista
    private List<T> c;

    // Constructor - Se genera una lista vacia
    public Cola(List<T> cola)
    {
        this.c = cola;
    }

    // encolar - Se agrega un elemento en la cola
    public void encolar(T elemento)
    {
        this.c.Add(elemento);
    }

    // desencolar - Se elimina el primer elemento de la cola
    public T desencolar()
    {
        T elementoTemp = this.c[0];
        this.c.RemoveAt(0);
        return elementoTemp;
    }

    // esVacia - Se verifica si la cola es vacia o no
    public bool esVacia()
    {
        return this.c.Count() == 0;
    }

    // getPrimerDato - Se devuelve el valor del primer dato
    public T getPrimerDato()
    {
        return this.c[0];
    }
}

File: Cola/Cola.py
# Clase Cola - Primero que entra, primero que sale

class Cola:
    
    # Consutructor - Se genera una lista vacia
    def __init__(self):
        self.__c = []
    
    # Metodos
    
    # encolar - Se agrega un elemento en la cola
    def encolar(self, dato):
        self.__c.append(dato)
    
    # desencolar - Se elimina el primer elemento de la cola    
    def desencolar(self):
        return self.__c.pop(0)
    
    # esVacia - Se verifica si la cola es vacia o no
    def esVacia(self):
        return len(self.__c) == 0
        
    # getPrimerDato - Se devuelve el valor del primer dato
    def getPrimerDato(self):
        return self.__c[0]